### PR TITLE
feat: add sdk password and pin checks

### PR DESCRIPTION
**Pull request by kristifidani**
## Motivation and Context

Explain why this change is necessary or the issue it addresses.

## Summary

- Added checks for sdk password with this crate: https://crates.io/crates/zxcvbn. the changes are only in the `newtypes.rs` file and then they affected ~100 more files xD.
- Added check for pin to be min 6 digits.
- Updated all tests and examples that were affected. (almost everything 😅 and that's why there are many file changes)
- For wasm bindings put the `pin` in the `utils` instead of having it hardcoded in every example. 
- Hardcode the wallet password. We were using the same password for authenticating the user and wallet. That does not work anymore so we needed to update all examples.

Backend changes: https://gitlab.inovel.de/zd0006/cryptpay.rs/-/merge_requests/537

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [ ] Code follows project style and guidelines.
- [ ] No redundant or duplicate code.
- [ ] No added new dependencies without clear justification and approval.

### Documentation

- [ ] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [ ] Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.
- [ ] Tests pass locally.

### Build & CI/CD

- [ ] Confirmed no CI/CD pipeline issues.
- [ ] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

File: sdk/src/types/newtypes.rs [hunk 1]
```diff
@@ -26,8 +26,11 @@ use aes_gcm::{
 };
 use iota_sdk::crypto::hashes::blake2b::Blake2b256;
 use iota_sdk::crypto::hashes::Digest;
+use log::warn;
 use rand::RngCore;
 use serde::{Deserialize, Serialize};
+extern crate zxcvbn;
```
Comment (antbern) [inline sdk/src/types/newtypes.rs:32]: You should not need the `extern crate` stuff (this is old!)
```suggestion
```

File: sdk/src/types/newtypes.rs [hunk 2]
```diff
@@ -152,6 +165,14 @@ impl EncryptionPin {
         if pin.is_empty() {
             return Err(TypeError::EmptyPin);
         }
+
+        if pin.len() < 6 {
+            warn!("Pin is less than 6 digits");
+            return Err(TypeError::WeakPin);
+        }
+
+        // TODO: check numerical values?
```
Comment (antbern) [inline sdk/src/types/newtypes.rs:174]: So what do we do with this in the end?

File: sdk/src/types/newtypes.rs [hunk 3]
```diff
@@ -52,6 +55,15 @@ impl PlainPassword {
...
         }
 
+        // Validates the strength of the password.
+        // Any score less than 3 (can be cracked with 10^10 guesses or less) should be considered too weak. [Score](https://docs.rs/zxcvbn/latest/zxcvbn/enum.Score.html)
+        let password_strength = zxcvbn(password.as_str(), &[]).score();
+
+        if password_strength < Score::Three {
+            warn!("User attempted to set a weak password");
+            return Err(TypeError::WeakPassword);
+        }
```
Comment (antbern) [inline sdk/src/types/newtypes.rs:65]: Can we add some unit-tests that makes sure that to simple passwords fail? Same with the pin also 🌵

**Review by antbern — COMMENTED** [review]
lots of changes! 🥇

File: sdk/src/types/newtypes.rs (hunk 2 above)
Comment (kristifidani) [reply sdk/src/types/newtypes.rs]: Added check for that too 👍

File: sdk/src/types/newtypes.rs (hunk 3 above)
Comment (kristifidani) [reply sdk/src/types/newtypes.rs:64]: Added some test cases 👍

**Review by antbern — APPROVED** [review]

**Review by antbern — APPROVED** [review]

**Review by hauju — APPROVED** [review]
